event-schema/counter_manager_test.go

```diff
@@ -0,0 +1,51 @@
+package event_schema
```

Comment:
Can we have a `TestConcurrentGetFrequencyCounterBounded` where multiple go routines are invoking the function? Please remember to run it with the `-race` flag like `go test -race ...`.

Reply:
Can you at least create a Notion ticket to check the race conditions here? There are lots of accesses to the same variables (e.g. `countersCache`) but the mutexes that are supposed to protect access are not even in the same file. We should look into this.